src/neuroconv/datainterfaces/behavior/video/externalvideointerface.py

```diff
@@ -314,7 +294,29 @@ def add_to_nwbfile(
         image_series_kwargs = metadata["Behavior"]["ExternalVideo"][self.video_name]
         image_series_kwargs["name"] = self.video_name
 
-        timing_type = self.get_timing_type()
+        if always_write_timestamps:
+            timestamps = self._timestamps if self._timestamps is not None else self.get_timestamps()
+            image_series_kwargs.update(timestamps=np.concatenate(timestamps))
+        elif self._timestamps is not None:
+            # Check if timestamps are regular
+            timestamps = np.concatenate(self._timestamps)
+            rate = calculate_regular_series_rate(series=timestamps)
+            if rate is not None:
+                starting_time = timestamps[0]
+                image_series_kwargs.update(starting_time=starting_time, rate=rate)
+            else:
+                image_series_kwargs.update(timestamps=timestamps)
+        else:
+            if self._number_of_files > 1 and self._segment_starting_times is None:
+                raise ValueError(
+                    f"No timing information is specified and there are {self._number_of_files} total video files! "
+                    "Please specify the temporal alignment of each video."
+                )
+            starting_time = self._segment_starting_times[0] if self._segment_starting_times is not None else 0.0
```

Comment:
I realize this is from the old implementation, but what's the point of setting a list of `self._segment_starting_times` if only the first one ever gets used...?

Reply:
Specifically, I think it would be confusing for a user to specify a list of segment starting times, and then check the ImageSeries only to find it has just a single `starting_time` and `rate`.

Reply:
I find the current timing functions a bit confusing and would like to improve them but I think we can't get to the situation you propose because of this assert.

https://github.com/catalystneuro/neuroconv/blob/dc3b4a63d29fd5a8095c40754b3d2c94428b5e45/src/neuroconv/datainterfaces/behavior/video/externalvideointerface.py#L209-L213

Reply:
I have an idea: What if instead of raising a ValueError in the else here: https://github.com/catalystneuro/neuroconv/blob/dc3b4a63d29fd5a8095c40754b3d2c94428b5e45/src/neuroconv/datainterfaces/behavior/video/externalvideointerface.py#L163-L190

We set 
```python
self._segment_starting_times[0] = aligned_starting_time
```

And then we raise a ValueError in the else here: https://github.com/catalystneuro/neuroconv/blob/dc3b4a63d29fd5a8095c40754b3d2c94428b5e45/src/neuroconv/datainterfaces/behavior/video/externalvideointerface.py#L192-L224

Reply:
Then a user would only use `set_aligned_segment_starting_times` if they are aligning all the timestamps, and if they are using a starting_time and rate representation, they would use `set_aligned_starting_time`.

Reply:
> Hahaha maybe we should separate the ExternalFile into two lol

Yeah, that's not a bad idea, but I don't think we _really_ need to.

Reply:
> Would you like to create a PR with this as a base for that improvement?

Sure

Reply:
Ok, thanks a lot for that, @pauladkisson 

Anything else you think we should do or modify for this PR?